src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -633,7 +639,7 @@ def _evp_pkey_to_public_key(self, evp_pkey):
 
         key_type = self._lib.EVP_PKEY_id(evp_pkey)
 
-        if key_type == self._lib.EVP_PKEY_RSA:
+        if key_type in self._supported_rsa_key_types:
```

Comment:
What happens if you use a key that's got a type of `EVP_PKEY_RSA_PSS`? Does it do something weird?

Reply:
If it has PSS parameter restrictions OpenSSL will enforce them (e.g. hash alg, mask alg, salt length). Otherwise it behaves as a normal key.

Reply:
That sounds awful? It means you have to know where your key came from to know how `.sign()` is going to behave.

This doesn't seem like a good way to go.